build-system/tasks/report-test-results.js

```diff
@@ -20,11 +20,78 @@
 
 'use strict';
 
+const fetch = require('node-fetch');
+const fs = require('fs').promises;
 const log = require('fancy-log');
-const {green} = require('ansi-colors');
+const path = require('path');
+const {
+  travisBuildNumber,
+  travisJobNumber,
+  travisCommitSha,
+} = require('../common/travis');
+const {cyan, green, yellow} = require('ansi-colors');
+
+const REPORTING_API_URL = 'https://amp-test-cases.appspot.com/report';
+
+async function getReport(testType) {
+  const karmaReportString = await fs.readFile(
+    `result-reports/${testType}.json`
+  );
+  const karmaReportJson = JSON.parse(karmaReportString);
+
+  const travisReport = addJobAndBuildInfo(testType, karmaReportJson);
+
+  return travisReport;
+}
+
+function addJobAndBuildInfo(testType, reportJson) {
+  const build = {
+    buildNumber: travisBuildNumber(),
+    commitSha: travisCommitSha(),
+  };
+
+  const job = {
+    jobNumber: travisJobNumber(),
+    testSuiteType: testType,
+  };
+
+  return {build, job, results: reportJson};
+}
+
+async function sendTravisKarmaReport(testType) {
+  const body = await getReport(testType);
+
+  const response = await fetch(REPORTING_API_URL, {
+    method: 'post',
+    body: JSON.stringify(body),
+    headers: {'Content-Type': 'application/json'},
+  });
+
+  if (response.ok) {
+    log(
+      green('INFO:'),
+      `Test results of type`,
+      cyan(testType),
+      'reported to',
+      cyan(REPORTING_API_URL)
+    );
+  } else {
+    log(
+      yellow('WARNING:'),
+      'failed to report results of type',
+      cyan(testType),
+      ': \n',
+      yellow(response.statusText).substr(0, 100)
```

Comment:
The color functions work by wrapping text in [ANSI color codes](https://en.wikipedia.org/wiki/ANSI_escape_code#Colors), like `ESC[31m`. There are "open" and "close" tags, so by substring-ing the colorized contents, you may be cutting off the closing color tag. You probably want to substring the `statusText`.

On that note, why substring at all? AFAIK it's never more than a couple of words

Reply:
I took the idea from https://github.com/ampproject/amphtml/blob/f8ee003f7239be737c03a6626b0a326c5e9aae6a/build-system/tasks/report-test-status.js -- I figured it would be better to limit the length of the response just in case. I'll remove it.

Reply:
Done removing it (if you think it'd be better to keep it, we can add it to statusText like you suggested).